registry/endpoint.go

```diff
@@ -260,3 +261,14 @@ HeaderLoop:
 
 	return RegistryInfo{}, fmt.Errorf("v2 registry endpoint returned status %d: %q", resp.StatusCode, http.StatusText(resp.StatusCode))
 }
+
+func (e *Endpoint) HTTPClient() *http.Client {
+	return &http.Client{
+		Transport: &http.Transport{
+			DisableKeepAlives: true,
+			Proxy:             http.ProxyFromEnvironment,
+			TLSClientConfig:   &tls.Config{InsecureSkipVerify: !e.IsSecure},
```

Comment:
While we're doing this, do we need to also handle setting a root cert pool and list of client certificates? It looks like this is left out here. Do you think we could use the existing `newClient` function here?